.github/workflows/test_security.yml

```diff
@@ -0,0 +1,96 @@
+name: Security test workflow for k-NN
+on:
+  push:
+    branches:
+      - "*"
+      - "feature/**"
+  pull_request:
+    branches:
+      - "*"
+      - "feature/**"
+
+jobs:
+  Build-ad:
+    strategy:
+      matrix:
+        java: [ 11,17 ]
+        os: [ubuntu-latest]
```

Comment:
should we add mac and windows?

Reply:
I'm not sure that will benefit us as running on at least one platform gives lot of confidence, also it's not easy due to docker setup on those platforms. We can add later if needed